aws/src/main/java/org/apache/iceberg/aws/glue/IcebergToGlueConverter.java

```diff
@@ -0,0 +1,310 @@
+        "because it must be 1-255 chars of lowercase letters, numbers, underscore", tableName);
+  }
+
+  /**
+   * Validate and get Glue table name from Iceberg TableIdentifier
+   * @param tableIdentifier table identifier
+   * @return table name
+   */
+  static String getTableName(TableIdentifier tableIdentifier) {
+    validateTableName(tableIdentifier.name());
+    return tableIdentifier.name();
+  }
+
+  /**
+   * Validate Iceberg TableIdentifier is valid in Glue
+   * @param tableIdentifier Iceberg table identifier
+   */
+  static void validateTableIdentifier(TableIdentifier tableIdentifier) {
+    validateNamespace(tableIdentifier.namespace());
+    validateTableName(tableIdentifier.name());
+  }
+
+  /**
+   * Set Glue table input information based on Iceberg table metadata.
+   * <p>
+   * A best-effort conversion of Iceberg metadata to Glue table is performed to display Iceberg information in Glue,
+   * but such information is only intended for informational human read access through tools like UI or CLI,
+   * and should never be used by any query processing engine to infer information like schema, partition spec, etc.
+   * The source of truth is stored in the actual Iceberg metadata file defined by the metadata_location table property.
+   * @param tableInputBuilder Glue TableInput builder
+   * @param metadata Iceberg table metadata
+   */
+  static void setTableInputInformation(TableInput.Builder tableInputBuilder, TableMetadata metadata) {
+    try {
+      tableInputBuilder
+          .storageDescriptor(StorageDescriptor.builder()
+              .location(metadata.location())
+              .columns(toColumns(metadata))
+              .build());
+    } catch (RuntimeException e) {
+      LOG.warn("Encountered unexpected exception while converting Iceberg metadata to Glue table information", e);
+    }
+  }
+
+  /**
+   * Converting from an Iceberg type to a type string that can be displayed in Glue.
+   * <p>
+   * Such conversion is only used for informational purpose,
+   * DO NOT reference this method for any actual data processing type conversion.
+   *
+   * @param type Iceberg type
+   * @return type string
+   */
+  private static String toTypeString(Type type) {
+    switch (type.typeId()) {
+      case BOOLEAN:
+        return "boolean";
+      case INTEGER:
+        return "int";
+      case LONG:
+        return "bigint";
+      case FLOAT:
+        return "float";
+      case DOUBLE:
+        return "double";
+      case DATE:
+        return "date";
+      case TIME:
+      case STRING:
+      case UUID:
+        return "string";
+      case TIMESTAMP:
+        return "timestamp";
+      case FIXED:
+      case BINARY:
+        return "binary";
+      case DECIMAL:
+        final Types.DecimalType decimalType = (Types.DecimalType) type;
+        return String.format("decimal(%s,%s)", decimalType.precision(), decimalType.scale());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;precision&#39; expression, &#39;scale&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 235. Consider formatting this data with &#39;%d&#39; instead.